modules/freetype/include/opencv2/freetype.hpp

```diff
@@ -0,0 +1,166 @@
+//  copy or use the software.
+//
+//
+//                           License Agreement
+//                For Open Source Computer Vision Library
+//
+// Copyright (C) 2000-2008, Intel Corporation, all rights reserved.
+// Copyright (C) 2009-2012, Willow Garage Inc., all rights reserved.
+// Third party copyrights are property of their respective owners.
+//
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+//
+//   * Redistribution's of source code must retain the above copyright notice,
+//     this list of conditions and the following disclaimer.
+//
+//   * Redistribution's in binary form must reproduce the above copyright notice,
+//     this list of conditions and the following disclaimer in the documentation
+//     and/or other materials provided with the distribution.
+//
+//   * The name of the copyright holders may not be used to endorse or promote products
+//     derived from this software without specific prior written permission.
+//
+// This software is provided by the copyright holders and contributors "as is" and
+// any express or implied warranties, including, but not limited to, the implied
+// warranties of merchantability and fitness for a particular purpose are disclaimed.
+// In no event shall the Intel Corporation or contributors be liable for any direct,
+// indirect, incidental, special, exemplary, or consequential damages
+// (including, but not limited to, procurement of substitute goods or services;
+// loss of use, data, or profits; or business interruption) however caused
+// and on any theory of liability, whether in contract, strict liability,
+// or tort (including negligence or otherwise) arising in any way out of
+// the use of this software, even if advised of the possibility of such damage.
+//
+//M*/
+//################################################################################
+//
+//                    Created by Kumataro
+//
+//################################################################################
+
+#ifndef _OPENCV_FREETYPE_H_
+#define _OPENCV_FREETYPE_H_
+#ifdef __cplusplus
+
+#include <opencv2/core.hpp>
+
+#include <vector>
+#include <string>
+
+#include <ft2build.h>
+#include <freetype/freetype.h>
+#include <freetype/ftoutln.h>
+
+#include <hb.h>
+#include <hb-ft.h>
+
+/**
+@defgroup plot Plot function for Mat data
+*/
+
+namespace cv
+{
+    namespace freetype
+    {
+    //! @addtogroup freetype
+    //! @{
+
+        class CV_EXPORTS_W FreeType2
+        {
+            private:
+            FT_Library       mLibrary;
+            FT_Face          mFace;
+            FT_Outline_Funcs mFn;
+
+            std::vector < Point > mPts;
+
+            Point            mOrg;
+            int              mLine_type;
+            int              mThickness;
+            int              mHeight;
+            Scalar           mColor;
+            Mat              mImg;
```

Comment:
Probably you don't need to store "image" Mat here, it is enough to have reference to it: `Mat& mImg;`